From fda4b6a1a523926692c2589c80a9974c200ed647 Mon Sep 17 00:00:00 2001
From: Maximiliano Sandoval R <msandova@gnome.org>
Date: Fri, 29 Sep 2023 16:57:30 +0200
Subject: [PATCH 2/3] app: Loop over files

---
 src/application.rs | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/application.rs b/src/application.rs
index f17feb8..1c8f796 100644
--- a/src/application.rs
+++ b/src/application.rs
@@ -53,18 +53,18 @@ mod imp {
 
         fn open(&self, files: &[gio::File], _hint: &str) {
             debug!("Application::open");
-            files.iter().for_each(|f| {
+            for file in files {
                 let window = Window::new(&self.obj());
                 let group = gtk::WindowGroup::new();
                 group.add_window(&window);
+                window.present();
 
-                crate::spawn!(glib::clone!(@weak window, @strong f => async move {
-                    match window.open(&f).await {
-                        Ok(_) => window.present(),
-                        Err(err) => log::error!("Could not open {}: {err}", f.uri()),
+                crate::spawn!(glib::clone!(@strong window, @strong file => async move {
+                    if let Err(err) = window.open(&file).await {
+                        log::error!("Could not open {}: {err}", file.uri());
                     }
                 }));
-            });
+            }
         }
     }
 
-- 
2.41.0

